src/server/cluster/cluster_defs.cc

```diff
@@ -49,6 +49,26 @@ std::string MigrationInfo::ToString() const {
                       slot_ranges.ToString(), ")");
 }
 
+bool ClusterShardInfo::operator==(const ClusterShardInfo& r) const {
+  if (slot_ranges == r.slot_ranges && master == r.master) {
+    auto lreplicas = replicas;
+    auto lmigrations = migrations;
+    auto rreplicas = r.replicas;
+    auto rmigrations = r.migrations;
+    std::sort(lreplicas.begin(), lreplicas.end());
+    std::sort(lmigrations.begin(), lmigrations.end());
+    std::sort(rreplicas.begin(), rreplicas.end());
+    std::sort(rmigrations.begin(), rmigrations.end());
```

Comment:
Since the class is immutable, can we sort when we construct it? we can then `DCHECK` here that all is sorted, and otherwise make this comparison simpler

Reply:
yes it's possible, but I decided to write less because these vectors in most cases contain 0 or 1 elements. If you think that it's important I will rewrite this code

Reply:
I think that the code will be nicer like that (I didn't propose it for performance reasons), but it's not important :)